### ISSUE TITLE
TypeError: Cannot read property 'type' of undefined

### ISSUE THREAD

**Issue by crystalfp** [erikdesjardins/eslint-plugin-dollar-sign#12]
This morning, after adding typescript-eslint to eslint, VSCode (1.31.1) complains:
```
[Info  - 6:09:21 AM] ESLint server stopped.
[Info  - 6:09:22 AM] ESLint server running in node v10.2.0
[Info  - 6:09:22 AM] ESLint server is running.
[Info  - 6:09:24 AM] ESLint library loaded from: C:\Users\Mario Valle\AppData\Roaming\npm\node_modules\eslint\lib\api.js
[Error - 6:17:10 AM] ESLint stack trace:
[Error - 6:17:10 AM] TypeError: Cannot read property 'type' of undefined
    at shouldVarAssignmentStartWithDollar (C:\Users\Mario Valle\AppData\Roaming\npm\node_modules\eslint-plugin-dollar-sign\rules\dollar-sign.js:82:15)
    at checkVarDeclarations (C:\Users\Mario Valle\AppData\Roaming\npm\node_modules\eslint-plugin-dollar-sign\rules\dollar-sign.js:104:6)
    at checkVarDeclarations (C:\Users\Mario Valle\AppData\Roaming\npm\node_modules\eslint-plugin-dollar-sign\rules\dollar-sign.js:111:4)
    at checkVarDeclarations (C:\Users\Mario Valle\AppData\Roaming\npm\node_modules\eslint-plugin-dollar-sign\rules\dollar-sign.js:111:4)
    at checkVarDeclarations (C:\Users\Mario Valle\AppData\Roaming\npm\node_modules\eslint-plugin-dollar-sign\rules\dollar-sign.js:111:4)
    at Program:exit (C:\Users\Mario Valle\AppData\Roaming\npm\node_modules\eslint-plugin-dollar-sign\rules\dollar-sign.js:166:4)
    at listeners.(anonymous function).forEach.listener (C:\Users\Mario Valle\AppData\Roaming\npm\node_modules\eslint\lib\util\safe-emitter.js:45:58)
    at Array.forEach (<anonymous>)
    at Object.emit (C:\Users\Mario Valle\AppData\Roaming\npm\node_modules\eslint\lib\util\safe-emitter.js:45:38)
    at NodeEventGenerator.applySelector (C:\Users\Mario Valle\AppData\Roaming\npm\node_modules\eslint\lib\util\node-event-generator.js:251:26)

```
I don't see any problem with eslint, but it is annoying. Thanks for looking!

**Comment by erikdesjardins** [comment]
Can you tell what code is causing it?

It seems to work for some Typescript I've tried, e.g. [this file](https://github.com/rollup/rollup/blob/4ba9054c5614ffe28216784ecb2a44a94f6fd418/src/Chunk.ts) doesn't cause an error.

**Comment by crystalfp** [comment]
This consistently triggers the error. Rename it to kb-main.ts
[kb-main.txt](https://github.com/erikdesjardins/eslint-plugin-dollar-sign/files/2913822/kb-main.txt)
Thanks!

**Comment by crystalfp** [comment]
You need also types definition (again rename it to types.d.ts)
[types.d.txt](https://github.com/erikdesjardins/eslint-plugin-dollar-sign/files/2913842/types.d.txt)

**Comment by crystalfp** [comment]
The same problem happens with the normal eslint:
```
eslint -c eslint.json --color -f codeframe --env browser,amd,jquery kb-main.ts
```
[eslint.json](https://github.com/erikdesjardins/eslint-plugin-dollar-sign/files/2913866/eslint.txt)

**Comment by crystalfp** [comment]
Sorry for my fragmented comments. But now I have a very minimal example that triggers the error:
```
require([], function() {

	const ViewModel = (function() {

        const enum cfg {ENTRIES_PER_PAGE = 50};
    })();
});
```
If you comment the `const enum cfg` line the error disappears (and appears the normal eslint output)

**Comment by erikdesjardins** [comment]
Perfect, thanks--should be fixed in 1.0.2
